Source/WebCore/loader/ImageLoader.cpp

```diff
@@ -486,18 +487,32 @@ void ImageLoader::notifyFinished(CachedResource& resource, const NetworkLoadMetr
         return;
     }
 
-    if (hasPendingDecodePromises())
-        decode();
-    loadEventSender().dispatchEventSoon(*this, eventNames().loadEvent);
+    Ref doc = document();
+    setShouldDelayLoadEvent(doc, true);
+    m_image->protectedImage()->subresourcesAreFinished([this, protectedThis = Ref { *this }, doc]() mutable {
+        // It is technically possible state changed underneath us.
+        if (!m_image)
+            return;
+        setShouldDelayLoadEvent(doc, false);
```

Comment:
What happens for load event delay if `m_image` is null? Or if the image element goes to another document?

Reply:
I guess this would be solved by moving it into `subresourcesAreFinished`.